Client Integration tests can flicker

Occasionally one of the client integration test will fails with this error:
```
panic: 0x111A00868581f73AB42FEEF67D235Ca09ca1E8db, error in run loop: objective not approved

goroutine 2657 [running]:
log.(*Logger).Panic(0x103573bcc?, {0x14002721aa8?, 0x14002721ad8?, 0x2?})
	/opt/homebrew/Cellar/go/1.19.3/libexec/src/log/log.go:251 +0x68
github.com/statechannels/go-nitro/client/engine.(*Engine).Run(0x140005d6bb0)
	/Users/lagap/code/go-nitro/client/engine/engine.go:152 +0x5ac
created by github.com/statechannels/go-nitro/client.New
	/Users/lagap/code/go-nitro/client/client.go:47 +0x270
FAIL	github.com/statechannels/go-nitro/client_test	4.940s
FAIL
```

It sometimes occurs on CI and I've been able to reproduce it using `go test ./client_test -count=50 -shuffle=on`. 

It looks like the problem is that a proposal arriving before we receive a state. We handle creating the objective but we don't call into the policy maker like we do with [states](https://github.com/statechannels/go-nitro/blob/942ef1142bd25792b9b591ae59d83df33475f195/client/engine/engine.go#L201).

Here's the SNIPPED log file for the failing test:
```
0xBBB676: 13:22:03.202204 engine.go:258: Ignoring payload for complected objective  VirtualDefund-0x34874db9e5bc5929bb1b24154537af9827dcd643069d29a236b9a5672167895e
0x111A00: 13:22:03.202451 engine.go:499: Objective VirtualDefund-0x27c1061139f68b440d2f7ed372f34c7f152ab33b99c007d8770d1819ea72b9fe is WaitingForCompleteLedgerDefunding
0xBBB676: 13:22:03.203340 engine.go:499: Objective VirtualDefund-0x33f2d2d0b9fc66ad7d57b9a14e411aac43090546c441a0f90294a04ed51ae73c is WaitingForNothing
0xBBB676: 13:22:03.203443 engine.go:678: Sending message {To:0x111A00868581f73AB42FEEF67D235Ca09ca1E8db PayloadSummaries:[] ProposalSummaries:[{ObjectiveId:VirtualDefund-0x33f2d2d0b9fc66ad7d57b9a14e411aac43090546c441a0f90294a04ed51ae73c LedgerId:0x6650a8de641a6914678e86acf29447889ef79b1093c60612327b2a34535ed027 ProposalType:RemoveProposal TurnNum:8}] Payments:[] RejectedObjectives:[]}
0xBBB676: 13:22:03.203443 engine.go:160: Objective VirtualDefund-0x33f2d2d0b9fc66ad7d57b9a14e411aac43090546c441a0f90294a04ed51ae73c is complete & returned to API
0x111A00: 13:22:03.203814 engine.go:499: Objective VirtualDefund-0xbb8b313d77fbec78ec510d79b2b4514dc119859bbda0ef00c726abfc561c7199 is WaitingForCompleteLedgerDefunding
0xBBB676: 13:22:03.204104 engine.go:676: Receiving message {To:0xBBB676f9cFF8D242e9eaC39D063848807d3D1D94 PayloadSummaries:[] ProposalSummaries:[{ObjectiveId:VirtualDefund-0x34874db9e5bc5929bb1b24154537af9827dcd643069d29a236b9a5672167895e LedgerId:0x6650a8de641a6914678e86acf29447889ef79b1093c60612327b2a34535ed027 ProposalType:RemoveProposal TurnNum:7} {ObjectiveId:VirtualDefund-0x33f2d2d0b9fc66ad7d57b9a14e411aac43090546c441a0f90294a04ed51ae73c LedgerId:0x6650a8de641a6914678e86acf29447889ef79b1093c60612327b2a34535ed027 ProposalType:RemoveProposal TurnNum:8} {ObjectiveId:VirtualDefund-0xbb8b313d77fbec78ec510d79b2b4514dc119859bbda0ef00c726abfc561c7199 LedgerId:0x6650a8de641a6914678e86acf29447889ef79b1093c60612327b2a34535ed027 ProposalType:RemoveProposal TurnNum:9}] Payments:[] RejectedObjectives:[]}
0xBBB676: 13:22:03.204416 engine.go:258: Ignoring payload for complected objective  VirtualDefund-0x34874db9e5bc5929bb1b24154537af9827dcd643069d29a236b9a5672167895e
0xBBB676: 13:22:03.204697 engine.go:258: Ignoring payload for complected objective  VirtualDefund-0x33f2d2d0b9fc66ad7d57b9a14e411aac43090546c441a0f90294a04ed51ae73c
0x111A00: 13:22:03.205086 engine.go:499: Objective VirtualDefund-0x33f2d2d0b9fc66ad7d57b9a14e411aac43090546c441a0f90294a04ed51ae73c is WaitingForCompleteLedgerDefunding
0xBBB676: 13:22:03.205870 engine.go:499: Objective VirtualDefund-0xbb8b313d77fbec78ec510d79b2b4514dc119859bbda0ef00c726abfc561c7199 is WaitingForNothing
0xBBB676: 13:22:03.205978 engine.go:160: Objective VirtualDefund-0xbb8b313d77fbec78ec510d79b2b4514dc119859bbda0ef00c726abfc561c7199 is complete & returned to API
0xBBB676: 13:22:03.205991 engine.go:678: Sending message {To:0x111A00868581f73AB42FEEF67D235Ca09ca1E8db PayloadSummaries:[] ProposalSummaries:[{ObjectiveId:VirtualDefund-0xbb8b313d77fbec78ec510d79b2b4514dc119859bbda0ef00c726abfc561c7199 LedgerId:0x6650a8de641a6914678e86acf29447889ef79b1093c60612327b2a34535ed027 ProposalType:RemoveProposal TurnNum:9}] Payments:[] RejectedObjectives:[]}
0x111A00: 13:22:03.206723 engine.go:499: Objective VirtualDefund-0x34874db9e5bc5929bb1b24154537af9827dcd643069d29a236b9a5672167895e is WaitingForNothing
0x111A00: 13:22:03.206821 engine.go:678: Sending message {To:0xAAA6628Ec44A8a742987EF3A114dDFE2D4F7aDCE PayloadSummaries:[] ProposalSummaries:[{ObjectiveId:VirtualDefund-0x34874db9e5bc5929bb1b24154537af9827dcd643069d29a236b9a5672167895e LedgerId:0xc4b7ff53f16db8148a1ac4f64caac5b865174043251af210f9285a5930e42c79 ProposalType:RemoveProposal TurnNum:10}] Payments:[] RejectedObjectives:[]}
0x111A00: 13:22:03.208536 engine.go:499: Objective VirtualDefund-0x722173bf67c6576fd82d11e172557fb57a404d97385dff2b42b2c69c60c41c81 is WaitingForCompleteLedgerDefunding
0x111A00: 13:22:03.208592 engine.go:678: Sending message {To:0xAAA6628Ec44A8a742987EF3A114dDFE2D4F7aDCE PayloadSummaries:[] ProposalSummaries:[{ObjectiveId:VirtualDefund-0x722173bf67c6576fd82d11e172557fb57a404d97385dff2b42b2c69c60c41c81 LedgerId:0xc4b7ff53f16db8148a1ac4f64caac5b865174043251af210f9285a5930e42c79 ProposalType:RemoveProposal TurnNum:11}] Payments:[] RejectedObjectives:[]}
0x111A00: 13:22:03.208612 engine.go:160: Objective VirtualDefund-0x34874db9e5bc5929bb1b24154537af9827dcd643069d29a236b9a5672167895e is complete & returned to API
0x111A00: 13:22:03.209300 engine.go:152: 0x111A00868581f73AB42FEEF67D235Ca09ca1E8db, error in run loop: objective not approved

```

--- a/client/engine/engine.go
+++ b/client/engine/engine.go
@@ -177,6 +177,10 @@ func (e *Engine) handleProposal(proposal consensus_channel.Proposal) (EngineEven
 	if err != nil {
 		return EngineEvent{}, err
 	}
+	if obj.GetStatus() == protocols.Completed {
+		e.logger.Printf("Ignoring proposal for complected objective  %s", obj.Id())
+		return EngineEvent{}, nil
+	}
 	return e.attemptProgress(obj)
 }
 

